### DISCUSSION TITLE
Purpose of mongodb & migration tasks

### DISCUSSION THREAD

**Discussion by kierangirvan** [ludeknovy/jtl-reporter#66] [Q&A] [answered]
Hi,

We are currently implementing the jtlreporter within AWS, we have managed to run the entire service in Fargate and using AWS RDS for the postgres DB.  As we are unable to run mongodb using an RDS service we are going to run the task with a non-persistent storage, do you see any issues with this, or does anything within the MongoDB need to be persisted?

Also - what is the purpose of the migration container?  When we initially launch the service it runs and then immediately terminates, just wondering if and when this task needs to run.

Really love the tool - super useful!  Keep up the good work!

**Comment by ludeknovy** [comment, marked as answer]
Hi @kierangirvan!

The mongodb serves as persistent storage for the actual samples(eg: each line of csv file). Once all data saved it runs the aggregations pipelines to get the statistics from those data. The results of the the aggregations is saved to postgres db. 
The migration container is the same as the backed, its purpose is to start - check for unprocessed migrations, if any found it will run them and terminate. So in most cases it will only start and terminate immediately.

I though Amazon DocumentDB is compatible with mongo? Maybe that could be the way?

In case you make it work in AWS, would you mind sharing the tutorial for it? We could add it to the docs.

**Reply by ludeknovy** [reply]
@kierangirvan could you please mark it as answered if I provided you the information you wanted ?
